src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/RemoveAzureContainerRegistry.cs

```diff
@@ -12,53 +12,57 @@
+    [OutputType(typeof(bool))]
     public class RemoveAzureContainerRegistry : ContainerRegistryCmdletBase
     {
-        [Parameter(
-            Position = 0,
-            Mandatory = true,
-            ParameterSetName = NameResourceGroupParameterSet,
-            HelpMessage = "Resource Group Name.")]
-        [ResourceGroupCompleter()]
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Resource Group Name.")]
         [ValidateNotNullOrEmpty]
         public string ResourceGroupName { get; set; }
 
-        [Parameter(
-            Position = 1,
-            Mandatory = true,
-            ParameterSetName = NameResourceGroupParameterSet,
-            HelpMessage = "Container Registry Name.")]
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Container Registry Name.")]
         [Alias(ContainerRegistryNameAlias, RegistryNameAlias, ResourceNameAlias)]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
-        [Parameter(
-            Mandatory = true,
-            ParameterSetName = RegistryObjectParameterSet,
-            ValueFromPipeline = true,
-            HelpMessage = "Container Registry Object.")]
-        [ValidateNotNullOrEmpty]
+        [Parameter(Mandatory = true, ParameterSetName = RegistryObjectParameterSet, HelpMessage = "Container Registry Object.")]
+        [ValidateNotNull]
         public PSContainerRegistry Registry { get; set; }
 
+        [Parameter(Mandatory = true, ParameterSetName = ResourceIdParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "The container registry resource id")]
+        [ValidateNotNullOrEmpty]
+        [Alias(ResourceIdAlias)]
+        public string ResourceId { get; set; }
+
         public override void ExecuteCmdlet()
         {
             if (string.Equals(ParameterSetName, RegistryObjectParameterSet))
             {
                 ResourceGroupName = Registry.ResourceGroupName;
                 Name = Registry.Name;
             }
+            else if (MyInvocation.BoundParameters.ContainsKey("ResourceId") || !string.IsNullOrWhiteSpace(ResourceId))
+            {
+                string resourceGroup, registryName, childResourceName;
+                if(!ConversionUtilities.TryParseRegistryRelatedResourceId(ResourceId, out resourceGroup, out registryName, out childResourceName))
+                {
+                    WriteInvalidResourceIdError(InvalidRegistryResourceIdErrorMessage);
+                    return;
+                }
+
+                ResourceGroupName = resourceGroup;
+                Name = registryName;
+            }
 
             if (ShouldProcess(Name, "Remove Container Registry"))
             {
                 RegistryClient.DeleteRegistry(ResourceGroupName, Name);
+                WriteObject(true);
```

Comment:
@Jinhuafei the pattern for `Remove-*` cmdlets (or any cmdlets that don't return output) is to have a `PassThru` switch parameter that a user can provide if they want some sort of output, which is usually a `bool` or the object that they have modified. I think we should follow that pattern here for consistency. Here is an example of [`Remove-AzureRmLoadBalancer`](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Network/Commands.Network/LoadBalancer/RemoveAzureLoadBalancerCommand.cs#L62-L65) implementing this.

Reply:
fixed